src/pip/_internal/resolution/resolvelib/provider.py

```diff
@@ -143,6 +144,22 @@ def get_preference(
             identifier,
         )
 
+    def _get_constraint(self, identifier: str) -> Constraint:
+        with contextlib.suppress(KeyError):
+            return self._constraints[identifier]
```

Comment:
Is this not just doing

```
if identifier in self._constraints:
    return self._constraints[identifier]
```

If it is, then I prefer the explicit if test. If it's not, then it needs a comment explaining why it needs contextlib.

Reply:
Changed